cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -256,19 +302,53 @@ func (ks *kube2sky) handlePodUpdate(old interface{}, new interface{}) {
 
 func (ks *kube2sky) handlePodDelete(obj interface{}) {
 	if e, ok := obj.(*kapi.Pod); ok {
-		if e.Status.PodIP != "" {
-			name := buildDNSNameString(ks.domain, podSubdomain, e.Namespace, santizeIP(e.Status.PodIP))
-			ks.mutateEtcdOrDie(func() error { return ks.removeDNS(name) })
+		ks.deleteIPBasedPodHostName(e)
+		ks.deleteAnnotationBasedPodHostName(e)
+	}
+}
+
+func (ks *kube2sky) deleteIPBasedPodHostName(e *kapi.Pod) {
+	if e.Status.PodIP != "" {
+		name := buildDNSNameString(ks.domain, podSubdomain, e.Namespace, santizeIP(e.Status.PodIP))
+		ks.mutateEtcdOrDie(func() error { return ks.removeDNS(name) })
+	}
+}
+
+func (ks *kube2sky) deleteAnnotationBasedPodHostName(pod *kapi.Pod) {
+	hostName := ks.getAnnotationBasedPodHostName(pod)
+	if len(hostName) > 0 {
+		recordLabel := getHash(pod.Name)
+		recordKey := buildDNSNameString(hostName, recordLabel)
+		ptrRecordKey := generatePTRRecordKey(pod.Status.PodIP)
+		ks.mutateEtcdOrDie(func() error {
+			glog.V(2).Infof("Deleting A record: %v, with recordKey: %v\n", hostName, recordKey)
+			err := ks.removeDNS(recordKey)
+			if err != nil {
+				return err
+			}
+			glog.V(2).Infof("Deleting PTR record: %v\n", ptrRecordKey)
+			return ks.removeDNS(ptrRecordKey)
+		})
+	}
+}
+
+func (ks *kube2sky) getAnnotationBasedPodHostName(e *kapi.Pod) string {
```

Comment:
call this `getPodFQDN` - that way it's clear that the return is fully qualified and that when we move to fields instead of annotations you just update this function